### ISSUE TITLE
AbstractMethodError in dotify-formatter or dotify-common

### ISSUE THREAD

**Issue by josteinaj** [nlbdev/pipeline#95]
As of:  nlbdev/pipeline@bf899902605184d574827dc701e1f2e774b11f53 (branch `fix-nordic-migrator.new`)

In maven.log:
```
java.lang.AbstractMethodError: org.daisy.dotify.formatter.impl.RowGroupDataSource.split(I)Lorg/daisy/dotify/common/split/SplitResult;
	at org.daisy.dotify.common.split.SplitPointHandler.skipLeading(SplitPointHandler.java:294)
	at org.daisy.dotify.common.split.SplitPointHandler.trimLeading(SplitPointHandler.java:281)
	at org.daisy.dotify.formatter.impl.PageSequenceBuilder2.nextPageInner(PageSequenceBuilder2.java:176)
	at org.daisy.dotify.formatter.impl.PageSequenceBuilder2.nextPage(PageSequenceBuilder2.java:134)
	at org.daisy.dotify.formatter.impl.SheetDataSource.ensureBuffer(SheetDataSource.java:123)
	at org.daisy.dotify.formatter.impl.SheetDataSource.getRemaining(SheetDataSource.java:59)
	at org.daisy.dotify.formatter.impl.VolumeProvider.prepare(VolumeProvider.java:87)
	at org.daisy.dotify.formatter.impl.FormatterImpl.getVolumes(FormatterImpl.java:138)
	at org.daisy.dotify.formatter.impl.FormatterImpl.write(FormatterImpl.java:123)
	at org.daisy.dotify.engine.impl.LayoutEngineImpl.convert(LayoutEngineImpl.java:128)
	at org.daisy.pipeline.braille.dotify.calabash.impl.OBFLToPEFStep.run(OBFLToPEFStep.java:136)
	at com.xmlcalabash.runtime.XAtomicStep.run(XAtomicStep.java:389)
	at com.xmlcalabash.runtime.XCompoundStep.run(XCompoundStep.java:262)
	at com.xmlcalabash.runtime.XChoose.run(XChoose.java:135)
	at com.xmlcalabash.runtime.XPipeline.doRun(XPipeline.java:236)
	at com.xmlcalabash.runtime.XPipeline.run(XPipeline.java:136)
	at com.xmlcalabash.extensions.Eval.run(Eval.java:249)
	at org.daisy.pipeline.braille.common.calabash.impl.PxTransformStep.run(PxTransformStep.java:143)
	at com.xmlcalabash.runtime.XAtomicStep.run(XAtomicStep.java:389)
	at com.xmlcalabash.runtime.XCompoundStep.run(XCompoundStep.java:262)
	at com.xmlcalabash.runtime.XChoose.run(XChoose.java:135)
	at com.xmlcalabash.runtime.XPipeline.doRun(XPipeline.java:236)
	at com.xmlcalabash.runtime.XPipeline.run(XPipeline.java:136)
	at com.xmlcalabash.runtime.XPipelineCall.run(XPipelineCall.java:94)
	at com.xmlcalabash.runtime.XPipeline.doRun(XPipeline.java:236)
	at com.xmlcalabash.runtime.XPipeline.run(XPipeline.java:136)
	at com.xmlcalabash.runtime.XPipelineCall.run(XPipelineCall.java:94)
	at com.xmlcalabash.runtime.XCompoundStep.run(XCompoundStep.java:262)
	at com.xmlcalabash.runtime.XTry.run(XTry.java:98)
	at com.xmlcalabash.runtime.XPipeline.doRun(XPipeline.java:236)
	at com.xmlcalabash.runtime.XPipeline.run(XPipeline.java:136)
	at com.xmlcalabash.extensions.Eval.run(Eval.java:249)
	at com.xmlcalabash.runtime.XAtomicStep.run(XAtomicStep.java:389)
	at com.xmlcalabash.runtime.XCompoundStep.run(XCompoundStep.java:262)
	at com.xmlcalabash.runtime.XTry.run(XTry.java:98)
	at com.xmlcalabash.runtime.XCompoundStep.run(XCompoundStep.java:262)
	at com.xmlcalabash.runtime.XChoose.run(XChoose.java:135)
	at com.xmlcalabash.runtime.XForEach.run(XForEach.java:117)
	at com.xmlcalabash.runtime.XPipeline.doRun(XPipeline.java:236)
	at com.xmlcalabash.runtime.XPipeline.run(XPipeline.java:136)
	at com.xmlcalabash.runtime.XPipelineCall.run(XPipelineCall.java:94)
	at com.xmlcalabash.runtime.XPipeline.doRun(XPipeline.java:236)
	at com.xmlcalabash.runtime.XPipeline.run(XPipeline.java:136)
	at org.daisy.common.xproc.calabash.impl.CalabashXProcPipeline.run(CalabashXProcPipeline.java:242)
	at org.daisy.common.xproc.calabash.impl.CalabashXProcPipeline.run(CalabashXProcPipeline.java:185)
	at org.daisy.maven.xproc.pipeline.DaisyPipeline2.run(DaisyPipeline2.java:81)
	at org.daisy.maven.xproc.xprocspec.XProcSpecRunner.run(XProcSpecRunner.java:185)
	at org.daisy.maven.xproc.xprocspec.XProcSpecRunner.run(XProcSpecRunner.java:261)
	at org.daisy.pipeline.junit.AbstractXSpecAndXProcSpecTest.runXProcSpec(AbstractXSpecAndXProcSpecTest.java:52)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.ops4j.pax.exam.invoker.junit.internal.ContainerTestRunner.runChild(ContainerTestRunner.java:67)
	at org.ops4j.pax.exam.invoker.junit.internal.ContainerTestRunner.runChild(ContainerTestRunner.java:37)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:160)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:138)
	at org.ops4j.pax.exam.invoker.junit.internal.JUnitProbeInvoker.invokeViaJUnit(JUnitProbeInvoker.java:123)
	at org.ops4j.pax.exam.invoker.junit.internal.JUnitProbeInvoker.findAndInvoke(JUnitProbeInvoker.java:96)
	at org.ops4j.pax.exam.invoker.junit.internal.JUnitProbeInvoker.call(JUnitProbeInvoker.java:72)
	at org.ops4j.pax.exam.nat.internal.NativeTestContainer.call(NativeTestContainer.java:108)
	at org.ops4j.pax.exam.spi.reactors.EagerSingleStagedReactor.invoke(EagerSingleStagedReactor.java:109)
	at org.ops4j.pax.exam.junit.impl.ProbeRunner$2.evaluate(ProbeRunner.java:278)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
	at org.ops4j.pax.exam.junit.impl.ProbeRunner.run(ProbeRunner.java:112)
	at org.ops4j.pax.exam.junit.PaxExam.run(PaxExam.java:93)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
```

```
➜  pipeline git:(fix-nordic-migrator.new) make VERBOSE=1 SKIP_GROUP_EVAL_TARGET=true EVAL=eval check-modules/nlb/book-to-pef
eval 'bash .make/mvn-test.sh' $(dirname modules/nlb/book-to-pef/.test)
+ set -e
+ set -o pipefail
+ for arg in "$@"
+ cd modules/nlb/book-to-pef
+ eval mvn --settings '"/home/jostein/daisy-pipeline/pipeline/settings.xml"' '-Dworkspace="/home/jostein/daisy-pipeline/pipeline/.maven-workspace"' '-Dcache="/home/jostein/daisy-pipeline/pipeline/.maven-cache"' '-Dorg.ops4j.pax.url.mvn.localRepository="/home/jostein/daisy-pipeline/pipeline/.maven-workspace"' '-Dorg.daisy.org.ops4j.pax.url.mvn.settings="/home/jostein/daisy-pipeline/pipeline/settings.xml"' clean verify
++ mvn --settings /home/jostein/daisy-pipeline/pipeline/settings.xml -Dworkspace=/home/jostein/daisy-pipeline/pipeline/.maven-workspace -Dcache=/home/jostein/daisy-pipeline/pipeline/.maven-cache -Dorg.ops4j.pax.url.mvn.localRepository=/home/jostein/daisy-pipeline/pipeline/.maven-workspace -Dorg.daisy.org.ops4j.pax.url.mvn.settings=/home/jostein/daisy-pipeline/pipeline/settings.xml clean verify
+ eval tee -a /home/jostein/daisy-pipeline/pipeline/maven.log '|' cut -c1-1000 '|' pcregrep -M '"^\[INFO\]' '-+\n\[INFO\]' Building '.*\n\[INFO\]' '-+$|^\[(ERROR|WARNING)\]";' test '${PIPESTATUS[0]}' -eq 0
++ tee -a /home/jostein/daisy-pipeline/pipeline/maven.log
++ cut -c1-1000
++ pcregrep -M '^\[INFO\] -+\n\[INFO\] Building .*\n\[INFO\] -+$|^\[(ERROR|WARNING)\]'
[INFO] ------------------------------------------------------------------------
[INFO] Building DP2 Braille Modules :: NLB 1.10.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.16:test (default-test) on project mod-nlb: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/jostein/daisy-pipeline/pipeline/modules/nlb/book-to-pef/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
++ test 0 -eq 0
modules/nlb/book-to-pef/.deps.mk:6: recipe for target 'modules/nlb/book-to-pef/.test' failed
make: *** [modules/nlb/book-to-pef/.test] Error 1
```

[maven.log](https://github.com/nlbdev/pipeline/files/1333085/maven.log.txt)

[modules/nlb/book-to-pef/target/test.log](https://github.com/nlbdev/pipeline/files/1333095/test.log.txt)

@bertfrees: ideas?

**Comment by bertfrees** [comment]
Joel has updated some Dotify bundles to the latest version, dotify.common being one of them. Could have to do with that.

**Comment by josteinaj** [comment]
@joeha480 does the exception above ring any bell?

**Comment by josteinaj** [comment]
@bertfrees btw, I'm getting an error about modules/braille again, don't think it was there earlier:

```
➜  pipeline git:(fix-nordic-migrator.new) git subrepo-status modules/braille  
Smartmatch is experimental at .git-utils/git-subrepo-status line 215.
modules/brailleERROR: corrupt git-subrepo commit: 3a79395304401b0523b8c1a5b1f01f8d08eef810 (commit 73f56cc540 does not exist: might have been purged, or was never pushed)
 ++39
```

Not sure if it matters...

**Comment by bertfrees** [comment]
Oh right, I was gonna look at this error.

Regarding the subrepo error: sorry I thought I had pushed it, but maybe it was rejected. Now it is pushed.

**Comment by josteinaj** [comment]
Ok, thanks. I had to do a `git fetch --quiet git@github.com:daisy/pipeline-mod-braille.git super/nlb`, but that's fine.

**Comment by bertfrees** [comment]
I have reverted the Dotify update on the NLB branch. Like I suspected there is indeed some incompatibility between the latest dotify.common and the NLB branch of dotify.formatter.impl. This also means that one of the PRs https://github.com/brailleapps/dotify.formatter.impl/pull/29 or https://github.com/brailleapps/dotify.formatter.impl/pull/28 has a problem when combined with the latest dotify.common.
